tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -289,6 +294,38 @@ TEST_F(ChargepointTestFixtureV201, K01FR19_NumberPhasesOtherThan1AndPhaseToUseSe
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodInvalidPhaseToUse));
 }
 
+TEST_F(ChargepointTestFixtureV201, K01FR20_IfPhaseToUseSetAndACPhaseSwitchingSupportedUndefined_ThenProfileIsInvalid) {
+    auto new_device_model = create_device_model("file:device_model2?mode=memory&cache=shared", {});
+    device_model = std::move(new_device_model);
+
+    auto periods = create_charging_schedule_periods_with_phases(0, 1, 1);
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), uuid(),
+        ChargingProfileKindEnum::Absolute, 1);
+
+    auto sut = handler.validate_profile_schedules(profile);
+
+    EXPECT_THAT(sut,
+                testing::Eq(ProfileValidationResultEnum::ChargingSchedulePeriodPhaseToUseACPhaseSwitchingUnsupported));
+}
+
+TEST_F(ChargepointTestFixtureV201, K01FR20_IfPhaseToUseSetAndACPhaseSwitchingSupportedFalse_ThenProfileIsInvalid) {
+    auto new_device_model = create_device_model("file:device_model2?mode=memory&cache=shared", "false");
```

Comment:
It took me a few reads before I realized that this line was where the sole difference was between the two tests. I wonder if there can be a refactoring to make it more obvious?

Reply:
Renamed `new_device_model` to something that's more descriptive for each test.